lib/vue/src/components/map/CMap.vue

```diff
@@ -0,0 +1,307 @@
+        :lat-lng="marker"
+        :icon="getIcon(marker.color)"
+        :opacity="activeMarkerIndex === undefined || i == activeMarkerIndex ? 1.0 : 0.6"
+        @click="onMarkerClick(i, marker)"
+      />
+    
+      <l-control class="leaflet-bar">
+        <a
+          v-if="!hideCurrentLocationButton"
+          :title="labels.tooltip && labels.tooltip.goToCurrentLocation"
+          role="button"
+          class="d-flex justify-content-center align-items-center"
+          @click="goToCurrentLocation"
+        >
+          <font-awesome-icon
+            :icon="['fas', 'location-arrow']"
+            class="text-primary"
+          />
+        </a>
+      </l-control>
+    </l-map>
+  </div>
+</template>
+
+<script>
+import { OpenStreetMapProvider } from 'leaflet-geosearch'
+import { divIcon } from 'leaflet'
+import { LControl, LPolygon } from 'vue2-leaflet'
+import CInputSearch from '../input/CInputSearch.vue'
+
+export default {
+  props: {
+    hideCurrentLocationButton: {
+      type: Boolean,
+      default: false
+    },
+    
+    labels: {
+      type: Object,
+      required: true
+    },
+    
+    map: {
+      type: Object
+    },
+    
+    markers: {
+      type: Array,
+      default: () => ([])
+    },
+
+    activeMarkerIndex:{
+      type: Number,
+      default: undefined
+    },
+
+    mapStyle: {
+      type: String,
+      default: "height: 75vh; width: 100%;"
+    },
+
+    polygons: {
+      type: Array,
+      default: () => ([])
+    },
+
+    hideGeoSearch: {
+      type: Boolean,
+      default: false
+    },
+
+    disable: {
+      type: Boolean,
+      default: false
+    }
+  },
+
+  components: {
+    LControl,
+    CInputSearch,
+    LPolygon
+  },
+  
+  data () {
+    return {
+      geoSearch: {
+        query: '',
+        provider: new OpenStreetMapProvider(),
+        results: [],
+      },
+    }
+  },
+
+  computed: {
+    mapOptions () {
+      const defaultOptions = {
+        show: true,
+        zoom: 3,
+        center: [30, 30],
+        rotation: 0,
+        attribution: '&copy; <a target="_blank" href="http://osm.org/copyright">OpenStreetMap</a>',
+      }
+
+      return {
+        ...defaultOptions,
+        ...this.map
+      }
+    }
+  },
+
+  watch: {
+    mapOptions: {
+      handler (map) {
+        if (map.show && this.$refs.map) {
+          this.$refs.map.mapObject.invalidateSize()
+        }
+      }
+    }
+  },
+
+  mounted () {
+    this.onBoundsUpdate(this.$refs.map.mapObject.getBounds())
+  },
+
+  methods: {
+    onGeoSearch (query) {
+      if (!query) {
+        this.geoSearch.results = []
+        return
+      }
+
+      this.geoSearch.provider.search({ query }).then(results => {
+        this.geoSearch.results = results.map(result => ({
+          ...result,
+          latlng: {
+            lat: result.raw.lat,
+            lng: result.raw.lon,
+          },
+        }))
+      }).catch(() => {
+        this.$emit('trigger-geosearch-error')
```

Comment:
this should be called `on-geosearch-error`.
In general the reusable component doesn't know how the parent will handle the event, so the name should just indicate when the event is triggered.